lockup-factory/src/lib.rs

```diff
@@ -124,6 +125,12 @@ impl LockupFactory {
             foundation_account = Some(self.foundation_account_id.clone());
         };
 
+        // Defaults to the whitelist account ID given on init call.
+        let mut staking_pool_whitelist_account_id: AccountId = self.whitelist_account_id.clone();
```

Comment:
to remove mut which can be a potential issue later can be just

let staking_pool_whitelist_account_id: AccountId = if let Some(account_id) = whitelist_account_id { account_id.into() } else { self.whitelist_account_id.clone() };